torch/optim/asgd.py

```diff
@@ -81,9 +84,9 @@ def _init_group(self, group, params_with_grad, grads, mus, axs, etas, state_step
                 state = self.state[p]
                 # State initialization
                 if len(state) == 0:
-                    state["step"] = torch.tensor(0.0)
-                    state["eta"] = torch.tensor(group["lr"])
-                    state["mu"] = torch.tensor(1.0)
+                    state["step"] = torch.zeros((), device=p.device)
```

Comment:
Just confirming--the reason we don't have these stay on CPU today when capturable=False is because they interact with params with foreach ops now and thus need to be on the same device, yes?